src/modm/platform/clock/sam_pmc/clockgen_impl.hpp.in

```diff
@@ -11,34 +11,89 @@
 
 namespace modm::platform {
 
-template <MasterClkSource src, MasterClkPrescaler pres>
+template <MainInternalFreq freq>
+void ClockGen::enableMainInternal() {
+	// Enable Main RC Oscillator and wait for it to stabilize
+	PMC->CKGR_MOR |= CKGR_MOR_KEY_PASSWD | CKGR_MOR_MOSCRCEN;
+	while (!(PMC->PMC_SR & PMC_SR_MOSCRCS));
+
+	// Change Main RC Oscillator frequency
+	PMC->CKGR_MOR =
+		(PMC->CKGR_MOR & ~(CKGR_MOR_MOSCRCF_Msk)) |
+		CKGR_MOR_MOSCRCF((uint32_t)freq) |
```

Comment:
But there's no advantage to pass the argument via template?

Reply:
There kind of is. If this is a template, it saves 16 bytes of text section compared to placing the function in a cpp file. However, while pulling this data for you I got lazy and instead of moving it out of `_impl.hpp` (it's not allowed here as a non-template symbol due to multiple definitions warning) I just added inline specifier and bam! It's 16 bytes less than a template (so 32bytes less than cpp). Do you want me to make all of it `inline static`?

Reply:
> Do you want me to make all of it `inline static`?

Yes. The compiler will also aggressively optimize these inlined functions if called with constants, so even computations and branches will be resolved (compilers got really good in recent years).

Reply:
Sure. Do you have any clue why templates with "hardcoded" values are taking *more* space than regular inlined functions?

Reply:
I assume they get hit with different passes of the optimizer, perhaps also limited by ABI? I found that quite surprising too.